addons/survey/static/src/js/survey_session_chart.js

```diff
@@ -0,0 +1,245 @@
+                        }
+                    }]
+                },
+                tooltips: {
+                    enabled: false,
+                }
+            },
+        };
+    },
+
+    /**
+     * Returns the label of the associated survey.question.answer.
+     */
+    _extractChartLabels: function () {
+        return this.questionStatistics.map(function (point) {
+            return point.text.length > 20 ? point.text.substring(0, 17) + '...' : point.text;
+        });
+    },
+
+    /**
+     * We simply return an array of zeros as initial value.
+     * The chart will update afterwards as attendees add their user inputs.
+     */
+    _extractChartData: function () {
+        return this.questionStatistics.map(function () {
+            return 0;
+        });
+    },
+
+    /**
+     * Custom method that returns a color from SESSION_CHART_COLORS.
+     * It loops through the ten values and assign them sequentially.
+     *
+     * We have a special mechanic when the host shows the answers of a question.
+     * Wrong answers are "faded out" using a 0.3 opacity.
+     *
+     * @param {Object} metaData
+     * @param {Integer} metaData.dataIndex the index of the label, matching the index of the answer
+     *   in 'this.questionAnswers'
+     */
+    _getBackgroundColor: function (metaData) {
+        var opacity = '0.8';
+        if (this.showAnswers && this.questionHasAnswers) {
+            if (!this.questionAnswers[metaData.dataIndex]){
+                opacity = 0.3;
+            }
+        }
+        var rgb = SESSION_CHART_COLORS[metaData.dataIndex];
+        return `rgba(${rgb},${opacity})`;
+    },
+
+    /**
+     * Custom method that returns the survey.question.answer label color.
+     *
+     * Break-down of use cases:
+     * - Red if the host is showing answer, and the associated answer is not correct
+     * - Green if the host is showing answer, and the associated answer is correct
+     * - Black in all other cases
+     *
+     * @param {Object} metaData
+     * @param {Integer} metaData.dataIndex the index of the label, matching the index of the answer
+     *   in 'this.questionAnswers'
+     */
+    _getLabelColor: function (metaData) {
+        if (this.showAnswers && this.questionHasAnswers) {
+            if (this.questionAnswers[metaData.dataIndex]){
+                return '#2CBB70';
+            } else {
+                return '#D9534F';
+            }
+        }
+        return '#212529';
+    },
+
+    /**
+     * Special utility method that will process the statistics we receive from the
+     * survey.question#_prepare_statistics method.
+     *
+     * For multiple choice questions, the values we need are stored in a different place.
+     * We simply return the values to make the use of the statistics common for both simple and
+     * multiple choice questions.
+     *
+     * @param {Object} rawStatistics
+     */
+    _processQuestionStatistics: function (rawStatistics) {
```

Comment:
We could add a note "see _get_stats_data" as I suppose it comes from that method.